src/Compilers/CSharp/Portable/FlowAnalysis/EmptyStructTypeCache.cs

```diff
@@ -166,6 +166,14 @@ private FieldSymbol GetActualInstanceField(Symbol member, NamedTypeSymbol type)
                 switch (member.Kind)
                 {
                     case SymbolKind.Field:
+                        // Do not report virtual tuple fields.
+                        // They are additional aliases to the fields of the underlying struct or nested extensions.
+                        // and as such are already accounted for via the nonvirtual fields.
+                        if (field is TupleVirtualElementFieldSymbol)
```

Comment:
ShouldIgnoreStructField is a method emulating old compiler behavior for not other reasosn than compat and only if _dev12CompilerCompatibility is set. 

Ignoring virtual tuple fields has a different reason and is not conditional on a flag. I think we should not mix these.

---
In reply to: [74525116](https://github.com/dotnet/roslyn/pull/13107#discussion_r74525116) [](ancestors = 74525116,74524647)